plugin/vscode/extension.js

```diff
@@ -364,11 +371,14 @@ exports.DocumentLinterCollection = DocumentLinterCollection;
 
 let toDispose = [];
 
-async function activateAsync() {
+async function activateAsync(extensionContext = vscode.ExtensionContext) {
   let diagnostics = vscode.languages.createDiagnosticCollection();
   toDispose.push(diagnostics);
-
-  let linters = new DocumentLinterCollection(diagnostics);
+  let performanceWriter = new PerformanceWriter(extensionContext)
+  performanceWriter.createHeader()
+  setInterval(() => {performanceWriter.writeToFile()}, 2000);
```

Comment:
Blocking: We don't want to be waking up VS Code while the user is idle. We should only arm timers when we have something to write to the performance log.

Reply:
Resolved in latest commit